src/Components/Layout.tsx

```diff
@@ -1,596 +1,344 @@
+  DropdownData,
+  refreshData as rData,
+  RefreshData,
+  bodyConfig as bConfig,
+  BodyConfig,
+} from "./Types";
+import { style as themeStyle } from "./Style";
+
+interface Inputs {
+  onTimerEvent(): void;
+  onDateEvent(dates: Date[]): void;
+  dateFormatter?: Intl.DateTimeFormat;
+  theme?: any;
+  commonlyUsedText?: number[];
+  quickSelectTerms?: string[];
+  quickSelectIntervals?: number[];
+  relativeTerms?: string[];
+  relativeIntervals?: number[];
+  localeString?: string;
+  nowText?: string;
+  minimumYearValue?: number;
+  maximumYearValue?: number;
+  humanizer?: any;
+  setStoredRange?(dateRange: string[]): void;
+  storedRange: string[] | null;
+}
+
+export const Layout: React.FC<Inputs> = (props) => {
+  // GLOBAL
+  let defaults = useContext(GlobalContext);
+  defaults.classes = themeStyle();
+  defaults.commonlyUsedText =
+    props.commonlyUsedText || defaults.commonlyUsedText;
+
+  defaults.quickSelectIntervals =
+    props.quickSelectIntervals || defaults.quickSelectIntervals;
+
+  defaults.relativeIntervals =
+    props.relativeIntervals || defaults.quickSelectIntervals;
+
+  defaults.minimumYearValue =
+    props.minimumYearValue || defaults.minimumYearValue;
+
+  defaults.maximumYearValue =
+    props.maximumYearValue || defaults.maximumYearValue;
+
+  defaults.localeObj = {
+    localeString: props.localeString || defaults.localeObj.localeString,
+    dateFormatter: props.dateFormatter || defaults.localeObj.dateFormatter,
+    quickSelectTerms:
+      props.quickSelectTerms || defaults.localeObj.quickSelectTerms,
+    relativeTerms: props.relativeTerms || defaults.localeObj.relativeTerms,
+    nowText: props.nowText || defaults.localeObj.nowText,
+    humanizer: props.humanizer || defaults.localeObj.humanizer,
+  };
+
+  // STORAGE
+  const setStoredRange:
+    | ((dateRange: DateRange) => void)
+    | (() => null) = props.setStoredRange
+    ? (dateRange) => {
+        let dates: string[] = [
+          String(dateRange.finalDates[0].getTime()),
+          String(dateRange.finalDates[1].getTime()),
+        ];
+        for (let i = 0; i < dates.length; i++) {
+          if (dateRange.isAbsolute(i)) {
+            dates[i] = dates[i] + "A";
+          } else {
+            dates[i] = dateRange.relativeMS[i] + "R";
+          }
+        }
+        return props.setStoredRange!(dates);
+      }
+    : () => {
+        return null;
+      };
+
+  let storedRange = new DateRange(defaults.localeObj, uiData);
+  {
+    let data = props.storedRange;
+    if (data) {
+      let identifier = [data[0].slice(-1), data[1].slice(-1)];
+      data = [data[0].slice(0, -1), data[1].slice(0, -1)];
+      for (let p = 0; p < data.length; p++) {
+        if (identifier[p] === "A") {
+          let out = new Date(parseInt(data[p]));
+          storedRange.setDate(out, p);
+        } else {
+          storedRange.setRelative(parseInt(data[p]), p);
+        }
+      }
+      storedRange.applyChanges();
+    }
+  }
+
+  // STATE OBJECTS
```

Comment:
Any time you have sections like this, it's a pretty good indication that you probably need a separate component.  Not always, but pulling code out into separate, smaller modules is good thing to make mistakes doing, since you'll learn a lot more about the pros-and-cons of decomposition.

Reply:
Aren't all my state objects supposed to be defined in Layout.tsx?

Reply:
All of these different sections - GLOBAL, STATE, etc - are indications that there’s “context” that can be extracted out of a function and into sub elements.  And no, there’s no reason all of your state objects are supposed to be defined in Layout.tsx, in fact state objects should be defined in the component that uses them.